contrib/natsio/src/main/java/com/google/cloud/dataflow/contrib/natsio/NatsIO.java

```diff
@@ -0,0 +1,359 @@
+/*
+ * Copyright (C) 2015 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.dataflow.contrib.natsio;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.channels.ClosedByInterruptException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Properties;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+import org.nats.Connection;
+import org.nats.MsgHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.cloud.dataflow.sdk.coders.Coder;
+import com.google.cloud.dataflow.sdk.coders.KvCoder;
+import com.google.cloud.dataflow.sdk.coders.SerializableCoder;
+import com.google.cloud.dataflow.sdk.coders.StringUtf8Coder;
+import com.google.cloud.dataflow.sdk.io.Sink;
+import com.google.cloud.dataflow.sdk.io.UnboundedSource;
+import com.google.cloud.dataflow.sdk.options.PipelineOptions;
+import com.google.cloud.dataflow.sdk.transforms.PTransform;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.PInput;
+import com.google.common.base.Preconditions;
+
+/**
+ * <p>{@link NatsIO} provides an API to publish messages to and subscribe subjects on <A href="http://nats.io">NATS server</A>.
+ * 
+ * <p>NATS server is a high-performance and lightweight cloud native messaging system. A message is a simple string and can be exchanged between
+ * clients written in various programming languages.
+ * 
+ * <p>NatsIO API requires <A href="http://nats.io">NATS server</A> running and accessible from Dataflow workers prior to job submissions. Typically,
+ * NATS server is expected to run on <A href="https://cloud.google.com/compute/">Google Compute Engine</A>.
```

Comment:
Is this a hard requirement, or can NATS server run anywhere? What makes it need to be on GCE specifically?